medicines/api/src/substance.rs

```diff
@@ -1,12 +1,17 @@
-use crate::{pagination, pagination::PageInfo};
+use crate::{pagination, pagination::PageInfo, product::Product};
 
 #[derive(juniper::GraphQLObject)]
 #[graphql(description = "An active ingredient found in medical products")]
-struct Substance {
+pub struct Substance {
     name: String,
+    products: Option<Vec<Product>>,
 }
 
 impl Substance {
+    pub fn new(name: String, products: Option<Vec<Product>>) -> Self {
+        Self { name, products }
+    }
+
     #[allow(dead_code)]
```

Comment:
is this needed?

Reply:
an unfortunate consequence of the library we're using not being smart enough to recognise that this method is used